libs/SmartSync/src/com/salesforce/androidsdk/smartsync/model/Layout.java

```diff
@@ -0,0 +1,398 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.smartsync.model;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the layout of a Salesforce object.
+ *
+ * @author bhariharan
+ * @see <a href="https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_responses_record_layout.htm">https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_responses_record_layout.htm</a>
+ */
+public class Layout {
+
+    private static final String ID = "id";
+    private static final String LAYOUT_TYPE = "layoutType";
+    private static final String MODE = "mode";
+    private static final String SECTIONS = "sections";
+
+    private String id;
+    private String layoutType;
+    private String mode;
+    private List<LayoutSection> sections;
+    private JSONObject rawData;
+
+    /**
+     * Creates an instance of this class from its JSON representation.
+     *
+     * @param object JSON object.
+     * @return Instance of this class.
+     */
+    public static Layout fromJSON(JSONObject object) {
+        Layout layout = null;
+        if (object != null) {
+            layout = new Layout();
+            layout.rawData = object;
+            layout.id = object.optString(ID);
+            layout.layoutType = object.optString(LAYOUT_TYPE);
+            layout.mode = object.optString(MODE);
+            final JSONArray sections = object.optJSONArray(SECTIONS);
+            if (sections != null) {
+                layout.sections = new ArrayList<>();
+                for (int i = 0; i < sections.length(); i++) {
+                    final JSONObject section = sections.optJSONObject(i);
+                    if (section != null) {
+                        layout.sections.add(LayoutSection.fromJSON(section));
+                    }
+                }
+            }
+        }
+        return layout;
+    }
+
+    /**
+     * Returns the ID of this layout.
+     *
+     * @return ID of this layout.
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Returns the layout type of this layout.
+     *
+     * @return Layout type of this layout.
+     */
+    public String getLayoutType() {
```

Comment:
Should that return an enum type?

Reply:
Nope. This is coming from the response itself. Besides, we don't have an enum of layout types - we leave it up to the UI SDK doc for that with a link. That way more types can be added in the future and no client-side changes or mutations would be required.

Reply:
Is there a fixed set of possible values?

Reply:
Right now it's `Compact` or `Full` I think, but it's not explicitly called out in the UI API doc, nor is it an `enum` on the server. Besides, the user already knows this since the user passes this into his/her call (`layoutType` is an argument). Right now, this class is a 1:1 representation of the layout response and uses the same data types as mentioned in the UI API doc. I'd prefer not to deviate from that. If the server side doc for the API was an `enum` of a specified set of values, we could change it, but for now, I feel like it's best to mirror the UI API doc 1:1.

Reply:
Ok.